src/main/java/com/microsoft/jenkins/acr/service/AzureContainerRegistry.java

```diff
@@ -42,24 +45,24 @@ public String queueTaskRequest(String resourceGroupName,
                 .withOs(OS.fromString(request.getPlatform().getOs()))
                 .withArchitecture(Architecture.fromString(request.getPlatform().getArchitecture()))
                 .withVariant(Variant.fromString(request.getPlatform().getVariant()));
-        RegistryDockerTaskRunRequest.DefinitionStages.DockerTaskRunRequestStepAttachable attachable = this.getClient()
+        boolean pushable = request.getImageNames() != null && request.getImageNames().size() > 0;
+        Map<String, OverridingArgument> args = new HashMap();
+        for (BuildArgument arg : request.getBuildArguments()) {
+            args.put(arg.getKey(), new OverridingArgument(arg.getValue(), arg.isSecrecy()));
+        }
+        return this.getClient()
                 .containerRegistries()
                 .getByResourceGroup(resourceGroupName, acrName)
                 .scheduleRun()
                 .withPlatform(platformProperties)
                 .withDockerTaskRunRequest()
                 .defineDockerTaskStep()
-                .withDockerFilePath(request.getDockerFilePath());
-
-        if (request.getImageNames() == null || request.getImageNames().size() == 0) {
-            attachable = attachable.withPushDisabled();
-        } else {
-            attachable = attachable.withPushEnabled()
-                    .withImageNames(request.getImageNames());
-        }
-
-        attachable = request.isNoCache() ? attachable.withoutCache() : attachable.withCache();
-        return attachable.attach()
+                .withDockerFilePath(request.getDockerFilePath())
+                .withPushEnabled(pushable)
+                .withImageNames(request.getImageNames())
```

Comment:
Not depend on `pushEnabled` now?

Reply:
yes, the pushEnabled/pushDisabled also been merged into pushEnabled(bool)